--- a/src/vte.cc
+++ b/src/vte.cc
@@ -10489,19 +10489,12 @@ update_repeat_timeout (gpointer data)
          * reinstall a new one because we need to delay by the amount of time
          * it took to repaint the screen: bug 730732.
 	 */
-	if (g_active_terminals == nullptr) {
-		_vte_debug_print(VTE_DEBUG_TIMEOUT,
-				"Stopping update timeout\n");
-		update_timeout_tag = 0;
-		again = false;
-        } else {
                 update_timeout_tag =
                         g_timeout_add_full (G_PRIORITY_DEFAULT_IDLE,
                                             VTE_UPDATE_REPEAT_TIMEOUT,
                                             update_repeat_timeout, NULL,
                                             NULL);
                 again = true;
-	}
 
 	in_update_timeout = FALSE;
 
--- a/src/vtedefines.hh
+++ b/src/vtedefines.hh
@@ -84,8 +84,8 @@
 #define VTE_MAX_INPUT_READ		0x1000
 #define VTE_DISPLAY_TIMEOUT		10
 #define VTE_UPDATE_TIMEOUT		15
-#define VTE_UPDATE_REPEAT_TIMEOUT	30
-#define VTE_MAX_PROCESS_TIME		100
+#define VTE_UPDATE_REPEAT_TIMEOUT	(1000/60)
+#define VTE_MAX_PROCESS_TIME		(1000/60)
 #define VTE_CELL_BBOX_SLACK		1
 #define VTE_DEFAULT_UTF8_AMBIGUOUS_WIDTH 1
