diff --git a/lib/bluetooth-client.c b/lib/bluetooth-client.c
index 367bdc13..a0459832 100644
--- a/lib/bluetooth-client.c
+++ b/lib/bluetooth-client.c
@@ -230,16 +230,7 @@ device_list_uuids (const gchar * const *uuids)
 gboolean
 bluetooth_client_get_connectable(const char **uuids)
 {
-	int i, j;
-
-	for (i = 0; uuids && uuids[i] != NULL; i++) {
-		for (j = 0; j < G_N_ELEMENTS (connectable_uuids); j++) {
-			if (g_str_equal (connectable_uuids[j], uuids[i]))
-				return TRUE;
-		}
-	}
-
-	return FALSE;
+	return TRUE;
 }
 
 static const char *
